app/ready_to_battle.py

```diff
@@ -0,0 +1,24 @@
+def ready_to_battle(knights: dict) -> list[tuple]:

```

Comment:
it is a too big function, try to divide this into a few function with different responsibilities